### DISCUSSION TITLE
Ability to use a mask?

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#18] [Ideas]
I would be curious if we could use a black and white mask where the black areas are used to generate imagery.  This could create very unique results that when composited together with the original parts of the image.. 

Anyhow just a random thought ;)

**Comment by rkhamilton** [comment]
Let me stew on this. It may be possible, but I don't have a great understanding of what's going on in the pytorch code. You may have noticed that essentially every VQGAN+CLIP imlementation out there uses the exact same core algorithm for image generation. I assume it's because nobody but the original author really understands it!

That said, I do see a path to something like setting an "init_weight" mask instead of just a float. Essentially I imagine you could create a mask and apply it to the input video frame, and where the mask is white, set the image weight to be very high.

As an aside, in looking into this I discovered that the code repo that I started from in my project had made a very big experimental change to the way they handled init_weights. Basically, the way they handled the part of the style transfer that keeps the original content the same works radically differently from every other implementation. You commented a while ago that your were seeing results that differed from others, and this is surely why.

That said, the experimental version does produce pretty cool and different results, which I can't replicate with the original version. I'm running a bunch of tests comparing the two so I can decide how best to handle it. I think I'd like to make both methods available, but the effect of init_weight is very different between methods. E.g. with the original method (that I'm not using in my code right now) you need a weight of 1-10 to get a strong effect. With the current, alternative, method you get a strong effect from 0.1. Because it's testing video, it takes a long time to run evaluations.

Anyway, look for a note maybe in a couple days when I push v1.2.0 that will include the option for the original way to keep init_image content, and the new way.

**Reply by gateway** [reply]
I would love both options, I have been doing tests with both the code you wrote and the night app.. the one you have right now seems to get much more similar results to night than some others.. https://nightcafe.studio/ some how they get some pretty smooth single image results..
